### ISSUE TITLE
Default libcurand path fails

### ISSUE THREAD

**Issue by rggs** [microsoft/DeepSpeed#5772]
I'm trying to use Zero stage 3 with the Adam optimizer offloaded to CPU. However, when I try to initialize the optimizer, I get a build error:

```shell
c++ cpu_adam.o cpu_adam_impl.o -shared -lcurand -L/path/to/conda/env/lib/python3.9/site-packages/torch/lib -lc10 -ltorch_cpu -ltorch -ltorch_python -o cpu_adam.so
/usr/bin/ld: cannot find -lcurand
```
Upon checking, of course libcurand is not in that directory. It is, however, available at `/path/to/conda/env/lib` or `/usr/local/cuda-12/lib64`. But I can't specify those paths in something like `LD_LIBRARY_PATH` because the default behavior is just to use the PyTorch path:

https://github.com/microsoft/DeepSpeed/blob/83aa184351778f7eaf96ada070cb85dd3787b9c8/op_builder/builder.py#L791

Basically, I think there are three solutions, and I'm not sure which is best:
1. Reinstall PyTorch in a way that forces it to use the existing paths that already have libcurand
2. Reinstall PyTorch in a way that includes libcurand in the PyTorch path specified above
3. Add the ability to specify additional or alternative paths to DeepSpeed.
Any insight here would be appreciated!

**Comment by rggs** [comment]
## Update
I was able to get it to build by copying `libcurand.so` from the system cuda at `/usr/local/cuda-12.0/lib` to the path specified by `torch.utils.cpp_extension.CUDA_HOME`. It may be that this is simply a quirk of my system, but I still find it odd that `libcurand.so` wasn't installed along with the other cuda packages.

**Comment by zbaishancha** [comment]
hello, I'm experiencing the same problem, what is your exact solution? Thank you very much!

**Comment by zbaishancha** [comment]
@rggs

**Comment by zbaishancha** [comment]
I get it. In my system I need to do the following: `cp /usr/local/cuda-11.6/lib64/libcurand.so /opt/conda/lib/python3.8/site-packages/torch/lib`
